### ISSUE TITLE
Respect the user's shell variable

### ISSUE THREAD

**Issue by daniel-j-h** [junegunn/fzf.vim#229]
- Category
    - [x] Question
    - [ ] Bug
    - [ ] Suggestion
- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Vim
    - [ ] Vim
    - [x] Neovim

I'm using the fish shell but I'm setting

    set shell=/bin/bash

in my Neovim config. Using FZF it seems like the commands are getting executed under Fish. I noticed this since I started to have problems with FZF's autocompletion (sometimes cursors inserted only control command but nothing happened, sometimes even ctrl-k/j didn't work - can't reproduce outside of Neovim) which went away when starting Neovim with `SHELL=/bin/bash nvim`. I _think_ this might be due to how FZF picks up the user's shell here:

https://github.com/junegunn/fzf/blob/1acd2adce271ffaf0bfb303c6dc27fe2b03c9e04/src/util/util.go#L103

Should `fzf.vim` respect the user's shell variable?

**Comment by junegunn** [comment]
So, if I understood correctly,

1. `&shell` is different from `$SHELL` on Vim
2. And you want fzf to use the value of `&shell` instead of `$SHELL`

Am I correct?

**Comment by daniel-j-h** [comment]
Yes, I think the Neovim-specific setting should be preferred if that's possible.

**Comment by junegunn** [comment]
What do you mean by "FZF's autocompletion"? Are you using fzf in `:terminal`, or are your referring to the commands this project provides?

Can you provide the exact steps to reproduce the issue you're having?

**Comment by daniel-j-h** [comment]
Yes I can reproduce - here is the minimal config I could reduce it to:
```
set shell=/bin/bash

call plug#begin('~/.vim/plugged')

Plug 'junegunn/fzf', { 'dir': '~/.fzf', 'do': './install --bin' }
Plug 'junegunn/fzf.vim'
nnoremap <C-p> :FZF<CR>

call plug#end()
```

Steps to reproduce: open `nvim`, press `<C-p>`, enter `test`. Sometimes it works fine, looking like this:

![good](https://cloud.githubusercontent.com/assets/527241/19815597/7a40feb4-9cf8-11e6-9f16-7057bc3084ad.png)

But other times entering `test` does not filter the list of suggestions at all. Then pressing `up`, `up` prints the control characters, then pressing `<C-j>` adds newlines, as seen here:

![bad](https://cloud.githubusercontent.com/assets/527241/19815596/7a3c44f0-9cf8-11e6-9316-edd49229e7ac.png)

If I start `nvim` with `SHELL=/bin/bash` I can't reproduce the issue.

**Comment by junegunn** [comment]
I can't seem to reproduce the issue no matter how many times I try

```sh
SHELL=/usr/local/bin/fish nvim +'set shell=/bin/bash' +FZF
```

Are you running the latest version of neovim?

**Comment by daniel-j-h** [comment]
I'm using the latest Neovim from the official Ubuntu ppa. It's reporting:

    nvim --version
    NVIM 0.1.6-dev

**Comment by junegunn** [comment]
I looked into the relevant parts of the code but I can't understand how $SHELL can cause the problem.

You can see the command running in the terminal buffer by pressing `<CTRL-\><CTRL-N>` then `<CTRL-G>`.

**Comment by daniel-j-h** [comment]
Here is the command running:

![cmd](https://cloud.githubusercontent.com/assets/527241/19822592/1c83fb48-9d19-11e6-854a-fdfc6dc8b1a7.png)

when I open `env SHELL=/bin/bash nvim` I get a bash command and I can no longer reproduce the issue.

**Comment by junegunn** [comment]
If you examine the command and the content of the temporary file after $SHELL, you'll notice that there's no place $SHELL can affect the behavior of fzf. And the code you linked above is not used in this context.

**Comment by daniel-j-h** [comment]
So you're saying the behavior I'm seeing is from Neovim _not_ respecting my `shell` variable from the config but respecting the `SHELL` env var?
